### PR TITLE
generate name

### PR DESCRIPTION
**Pull request by negar-rostamzadeh**
#144 #11 the name generator

### CODE REVIEW

File: scripts/smart-dispatch [hunk 1]
```diff
@@ -128,6 +128,14 @@ def main():
                       }
 
     job_generator = job_generator_factory(queue, commands, command_params, CLUSTER_NAME, path_job)
+    ctr = 0
+    # generating default names per each jobs in each batch
+    for pbs in job_generator.pbs_list:
```
Comment (mgermain) [inline scripts/smart-dispatch]: for pbs_id pbs in enumerate(job_generator.pbs_list):

File: scripts/smart-dispatch [hunk 2]
```diff
@@ -128,6 +128,14 @@ def main():
...
+        proper_size_name = utils.jobname_generator(jobname, str(ctr))
```
Comment (mgermain) [inline scripts/smart-dispatch]: You should use `str()` inside your function `jobname_generator`

File: smartdispatch/utils.py [hunk 3]
```diff
@@ -6,6 +6,25 @@
...
+    -------
+    str
+    The cropped version of the string.	
+    '''
+    # 64 - 1 since the total length including -1 should be less than 64
+    if len(jobname) + len(job_id) > 63:
+        croped_string = '{}_{}'.format(jobname[:63 - len(job_id)], job_id)
+    else:
+        croped_string = '{}_{}'.format(jobname, job_id)
+        return croped_string
```
Comment (mgermain) [inline smartdispatch/utils.py]: This should not be indented. The function will not return anything in the case it enters the if.

File: smartdispatch/utils.py [hunk 4]
```diff
@@ -6,6 +6,26 @@
 from distutils.util import strtobool
 from subprocess import Popen, PIPE
 
+def jobname_generator(jobname, job_id_int):
```
Comment (mgermain) [inline smartdispatch/utils.py]: you don't have to put int in the name, keep the same name or at least update the docstring

File: smartdispatch/tests/test_utils.py [hunk 5]
```diff
@@ -35,6 +35,10 @@ def test_generate_uid_from_string():
     assert_equal(utils.generate_uid_from_string("same text"), utils.generate_uid_from_string("same text"))
     assert_true(utils.generate_uid_from_string("same text") != utils.generate_uid_from_string("sametext"))
 
+def test_jobname_generator():
```
Comment (mgermain) [inline smartdispatch/tests/test_utils.py]: did you even run the tests? that makes no sense whatsoever.
jobname_generator is not returning a length it's returning a string you need to use `len()`

**Comment by MarcCote** [comment]
@negar-rostamzadeh thank you for you PR.

**Pull request merged**